src/reference/asciidoc/aggregator.adoc

```diff
@@ -846,7 +846,9 @@ If the flag is set to true, then when the expiry callback is invoked, any unmark
 
 [IMPORTANT]
 =====
-When using a `MessageGroupStoreReaper`, with a common `MessageStore` for multiple correlating endpoints, ensure proper `CorrelationStrategy` is used when same `MessageStore` is used for multiple aggregators to ensure unique correlationkey is being assigned to the messages.
+Since _version 5.0_ Correlating endpoint's keep track of correlation key of message's processed by the endpoint.
```

Comment:
How about this instead?

> When a shared `MessageStore` is used for different correlation endpoints, it is necessary to configure a proper `CorrelationStrategy` to ensure uniqueness for group ids.
> Otherwise unexpected behavior may happen when one correlation endpoint may release or expire messages from others - messages with the same correlation key are stored in the same message group.

Let me know and I'll fix on merge

Reply:
Looks good. Thanks.

Reply:
The main my point here that the issue is not only related to the `MessageGroupStoreReaper` and expiration. The aggregator definitely polls the whole group from the store and may decide to release it.

So, you might be surprised that another your aggregator has somehow lost its messages, meanwhile the current one may produce some strange output container.

Gary once said that we are providing `region` option on the message store for this kind of "the same correlation key" scenarios. Right, we have it, but with this IMPORTANT message we should protect ourselves from the configuration when end-user is missing the `region` option or some other flag for distinguishing.

I still don't see how can we catch the problem in the distributed environment, therefore I'm pretty satisfied just having this message in the Reference Manual. 😄 

Merging...

Thank you, @meherzad , for your effort and patience during our review.
Looking forward for more from your side!